test/runner.c

```diff
@@ -165,21 +177,40 @@ frida_java_init_vm (JavaVM ** vm, JNIEnv ** env)
   create_java_vm = dlsym (vm_module, "JNI_CreateJavaVM");
   g_assert (create_java_vm != NULL);
 
+  n_options = 5;
+  if (optimization_enabled) {
+    n_options += 4;
+  } else {
+    n_options += 1;
+  }
+
+  options = malloc(n_options * sizeof(JavaVMOption));
```

Comment:
```suggestion
  options = g_new0 (JavaVMOption, n_options);
```

(Seems we were leaving `extraInfo` as random garbage, though I don't know if it's actually used.)

Reply:
The array is allocated differently depending on the mode of operation, though.

Reply:
Yeah I meant that this was an existing bug, where we didn't initialize the `extraInfo` field of each option.